chart/kubeapps/templates/kubeapps-frontend-config.yaml

```diff
@@ -57,36 +57,40 @@ data:
       # Ensure each cluster can be reached (should only be
       # used with an auth-proxy where k8s credentials never leave
       # the cluster). See clusters option.
-      {{- range .Values.clusters }}
+    {{- range .Values.clusters }}
       location ~* /api/clusters/{{ .name }} {
-        rewrite /api/clusters/{{ .name }}/(.*) /$1 break;
-        rewrite /api/clusters/{{ .name }} / break;
-
-      {{/* Helm returns a nil pointer error when accessing foo.bar if foo doesn't
+    {{- $apiServiceURL := default "https://kubernetes.default" .apiServiceURL }}
+    {{- $parsed := urlParse $apiServiceURL }}
+    {{- $baseURL := (printf "%s://%s" (get $parsed "scheme") (get $parsed "host")) }}
+    {{- $path := get $parsed "path" }}
```

Comment:
Yea, I am in favor of that idea! I will add a helper function that returns a dict like you suggest. That way the gory details of `urlParse` and concatenating the scheme w/ the host can be hidden away from view 😄 

I will also add a comment at this point in the file explaining why it's necessary.

Reply:
@absoludity done in [`ff2ae1c` (#2337)](https://github.com/kubeapps/kubeapps/pull/2337/commits/ff2ae1c778a25cfe2099c54a87cfbf5a908df15d).

Note my `toYaml` / `fromYaml` pattern, where I convert the dict I want into YAML so that I can call `fromYaml` on it in the caller to cast it _back_ to a dict that can then be assigned.